File: src/dto/task.py
import uuid

from pydantic import BaseModel


class CreateTaskPayload(BaseModel):
    name: str
    description: str | None
    workspace_id: int
    project_id: int | None


class TaskInWorkspace(BaseModel):
    id: uuid.UUID
    number: int
    name: str
    description: str | None
    project_id: int | None


class UpdateTaskPayload(BaseModel):
    id: uuid.UUID
    name: str
    description: str | None

File: front/src/stores/models/task.ts
export interface TaskModel {
  id: string,
  number: number,
  name: string,
  description: string | null,
  project_id: number | null,
}

export interface TaskCreateModel {
  name: string,
  description: string | null,
  project_id: number | null,
}

export interface TaskUpdateModel {
  task_id: string,
  name: string,
  description: string | null,
  project_id: number | null,
}
